lms/djangoapps/courseware/courses.py

```diff
@@ -462,30 +467,31 @@ def get_course_assignments(course_key, user, request, include_access=False):
     Each returned object is a namedtuple with fields: block_key, title, url, date, requires_full_access
     """
     store = modulestore()
-    all_course_dates = get_dates_for_course(course_key, user)
     assignments = []
-    for (block_key, date_type), date in all_course_dates.items():
-        if date_type != 'due' or block_key.block_type != 'sequential':
-            continue
-
-        try:
-            item = store.get_item(block_key)
-        except ItemNotFoundError:
-            continue
-
-        if not item.graded:
-            continue
-
-        requires_full_access = include_access and _requires_full_access(store, user, block_key)
-        title = item.display_name or _('Assignment')
-
-        url = None
-        assignment_released = not item.start or item.start < datetime.now(pytz.UTC)
-        if assignment_released:
-            url = reverse('jump_to', args=[course_key, block_key])
-            url = request and request.build_absolute_uri(url)
-
-        assignments.append(_Assignment(block_key, title, url, date, requires_full_access))
+    # Ideally this function is always called with a request being passed in, but because it is also
+    # a subfunction of `get_course_date_blocks` which does not require a request, we are being defensive here.
+    if request:
```

Comment:
Style nit: I generally prefer such early-exits to be explicit about what they are doing. Like, of the form:
```
if not request:
    return assignments  # or return []
```

Has the added bonus of keeping the "main path" less indented and thus it reads more like a main code path.